src/main/java/com/selfxdsd/todocli/TodoVisitor.java

```diff
@@ -102,25 +116,33 @@ public FileVisitResult visitFile(
     ) throws IOException {
         final String file = path.toString();
         if (file.endsWith(".java") || file.endsWith(".js")) {
-            final List<Todo> todos = parser.parse(file);
-
-            if (todos.size() > 0) {
-                log("Found {} TODOs in {}:", todos.size(), file);
-            }
-
-            for (int i = 0; i < todos.size(); i++) {
-                Todo todo = todos.get(i);
-
-                String suffix;
-                if (i == todos.size() - 1) {
-                    suffix = "\n";
-                } else {
-                    suffix = "";
+            service.submit(() -> {
```

Comment:
@criske So, for each source code file, we start a separate thread?

Reply:
Nope, as per service config, at most 3 threads, rest of the files will be queued.
`ExecutorService service = Executors.newFixedThreadPool(3)`